### PR TITLE
Job Scheduler Component

### PR DESCRIPTION
**Pull request by Zohair-coder**
This pull request includes the new lab_job_scheduler component.

### CODE REVIEW

File: components/lab-job-scheduler/backend/src/lab_job_scheduler/app.py [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+    )
+    return [ScheduledJob(**json.loads(document.json_value)) for document in documents]
+
+
+def get_job_from_job_input(job_schedule: ScheduledJobInput) -> ScheduledJob:
+    return ScheduledJob(
+        cron_string=job_schedule.cron_string,
+        job_input=job_schedule.job_input,
+        created=datetime.datetime.now().isoformat(),
+        job_id=str(uuid.uuid4()),
```
Comment (JNKielmann) [inline components/lab-job-scheduler/backend/src/lab_job_scheduler/app.py:204]: Use the same function to generate the id that is also used in contaxy:
```
from contaxy.utils import id_utils
id_utils.generate_short_uuid()
```

File: components/lab-job-scheduler/backend/src/lab_job_scheduler/job_deployer.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    """Runs all scheduled jobs."""
+    for project_id, jobs in cached_scheduled_jobs.items():
+        for job_id, job in jobs.items():
+            if is_due(job):
+                logger.info(f"Deploying job {job_id}")
+                deploy_job(job, component_manager, project_id)
+                update_last_run(job, component_manager,
+                                project_id, cached_scheduled_jobs)
+                update_next_run(job, component_manager,
+                                project_id, cached_scheduled_jobs)
```
Comment (JNKielmann) [inline components/lab-job-scheduler/backend/src/lab_job_scheduler/job_deployer.py:21]: Do this in one function to only have one DB update

File: components/lab-job-scheduler/backend/src/lab_job_scheduler/app.py [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+    return list(cached_scheduled_jobs[project_id].values())
+
+
+@app.get(
+    "/project/{project_id}/schedule/{job_id}",
+    summary="List a job schedule",
+    status_code=status.HTTP_200_OK,
+    responses={**CREATE_RESOURCE_RESPONSES},
+)
+def list_schedule(
```
[the remark this answers is not in the document]
Comment (Zohair-coder) [reply components/lab-job-scheduler/backend/src/lab_job_scheduler/app.py:107]: Resolved in ffb329e24f90c14a93e2b2e6b12d9625eb1e7d68.

File: components/lab-job-scheduler/backend/src/lab_job_scheduler/app.py (hunk 1 above)
Comment (Zohair-coder) [reply components/lab-job-scheduler/backend/src/lab_job_scheduler/app.py]: Resolved in 839b1c30aa0fb87be8805706d597dc5aa4f9c286

File: components/lab-job-scheduler/backend/src/lab_job_scheduler/job_deployer.py (hunk 2 above)
Comment (Zohair-coder) [reply components/lab-job-scheduler/backend/src/lab_job_scheduler/job_deployer.py]: Resolved in 8aebfcb9215024e06ee83502c46e9be9a263be8e

**Pull request merged**